### PR TITLE
Handle SIGTERM and SIGINT

### PR DESCRIPTION
**Pull request by sissbruecker**
Currently single-file-cli does not handle SIGTERM and SIGINT. SIGINT occurs when terminating the process though the CLI when pressing <kbd>Cmd/Ctrl+C</kbd>. SIGTERM occurs when running single-file as a child process in another app, and sending the SIGTERM signal to that process so that it can properly terminate itself.

The current behavior is:
- On SIGINT: Chromium does get shut down, but temporary profile folder is not removed
- ON SIGTERM: Chromium is not shut down, temporary profile folder is not removed

This adds listeners for both SIGINT and SIGTERM and runs `closeBrowser` in them, so that both Chromium is terminated and the profile folder is removed.

For testing SIGINT, just start single-file from the CLI and press <kbd>Cmd/Ctrl+C</kbd>. To observe whether the profile was deleted, something can be printed to the console in `closeBrowser`.

For testing SIGTERM I was using these scripts:

<details>
<summary>Deno</summary>

```js
const command = new Deno.Command(Deno.execPath(), {
  args: [
    "run",
    "--allow-read",
    "--allow-write",
    "--allow-run",
    "--allow-net",
    "single-file",
    "https://github.com/wagoodman/dive",
    "test.html",
  ],
  stdin: "piped",
  stdout: "piped",
  stderr: "piped",
});
const subprocess = await command.spawn();
subprocess.stdout.pipeTo(
  Deno.openSync("output", { write: true, create: true }).writable,
);
subprocess.stderr.pipeTo(
  Deno.openSync("output_err", { write: true, create: true }).writable,
);

setTimeout(() => {
  subprocess.kill("SIGTERM");
}, 1000);
```

Run as `deno run --allow-read --allow-write --allow-run test-deno.js`

</details>

<details>
<summary>Node</summary>

```js
import child_process from 'child_process';

const child = child_process.spawn('node', ['single-file-node.js', 'https://github.com/wagoodman/dive', 'test.html']);
child.stdout.pipe(process.stdout);
child.stderr.pipe(process.stderr);

setTimeout(() => {
  child.kill('SIGTERM')
}, 1000);
```

Run as: `node test-node.js`

</details>

### CODE REVIEW

File: lib/deno-polyfill.js [hunk 1]
```diff
@@ -80,7 +80,7 @@ const Command = DENO_RUNTIME_DETECTED ? Deno.Command : class Command {
 		return {
 			status: new Promise((resolve, reject) => {
 				child.on("exit", code => {
-					if (code === 0 || code === 143) {
+					if (code === 0 || code === 143 || code === 130) {
```
Comment (sissbruecker) [inline lib/deno-polyfill.js:83]: `130` occurs on `SIGINT`. We don't want to throw an error in that case, but instead allow the signal listeners to run.

**Comment by gildas-lormeau** [comment]
I need to do some tests but isn't this issue specific to Node.js?

File: lib/deno-polyfill.js (hunk 1 above)
Comment (gildas-lormeau) [reply lib/deno-polyfill.js:83]: I agree with this fix. I merged it manually.

**Comment by gildas-lormeau** [comment]
Ok, I am wrong about `ref()`. It's the other way around actually so I'm OK with the PR.

**Comment by sissbruecker** [comment]
Ctrl-C does stop Chrome, but it does not run `closeBrowser`. So I guess it's related to how the process is created in the shell, and Ctrl-C kills all spawned child processes as well.

**Comment by sissbruecker** [comment]
Thanks!

**Comment by gildas-lormeau** [comment]
You're welcome. I would like to thank you again for your contribution and the test examples. I've just published a new version with your fix.

**Pull request merged**